docs/developer_guide/docstrings.md

```diff
@@ -0,0 +1,290 @@
+
+Code:
+```{code-block} python
+"""
+.. attention::
+    attention!
+
+.. caution::
+    caution!
+
+.. danger::
+    danger!
+
+.. error::
+    error!
+
+.. hint::
+    hint!
+
+.. important::
+    important!
+
+.. note::
+    note!
+
+.. tip::
+    tip!
+
+.. warning::
+    warning!
+
+"""
+```
+
+````{admonition} Result
+:class: seealso
+
+```{attention}
+attention!
+```
+
+```{caution}
+caution!
+```
+
+```{danger}
+danger!
+```
+
+```{error}
+error!
+```
+
+```{hint}
+hint!
+```
+
+```{important}
+important!
+```
+
+```{note}
+note!
+```
+
+```{tip}
+tip!
+```
+
+```{warning}
+warning!
+```
+````
+`````
+
+`````{tab-item} More
+
+**Other reST Elements:**
+
+You can also use various other reST elements to enrich the documentation:
+
+- **Lists:** For bullet points or numbered lists.
+- **Code Blocks:** For including code snippets.
+- **Tables:** For tabular data.
+
+Refer to the [Sphinx & RST Cheat Sheet](https://sphinx-tutorial.readthedocs.io/cheatsheet/) for more examples and details.
+
+`````
+``````
+
+### Arguments
+
+Document parameters under the `Args` section. Each parameter should include its name, type, and a brief description. Type definitions cannot have a line break. Use indentation to break long parameter descriptions into multiple lines, ensuring they are clearly readable.
+
+Example:
+
+```python
+def example_function(param1, param2, param3):
+    """This function performs an example operation.
+
+    Args:
+        param1 (int): The first parameter.
+        param2 (Union[str, List[str], Dict[str, Any]]): The second parameter, which can be
+            a string, a list of strings, or a dictionary with string keys and any type of
+            values. This parameter is used to demonstrate a long type definition.
+        param3 (str): The third parameter, which has a long description that needs to be
+            broken into multiple lines for better readability. This parameter is used
+            to show how to write long descriptions and how to indent them properly in
+            the docstring.
+
+    Returns:
+        bool: True if successful, False otherwise.
+
+    Raises:
+        ValueError: If `param1` is not an integer.
+    """
+```
+
+### Return Types
+
+Document return types under the `Returns` section. Each return type should include the type and a brief description.
+
+Example:
+
+```python
+def example_function(param1, param2):
+    """This function performs an example operation.
+
+    Returns:
+        bool: True if successful, False otherwise.
+    """
+```
+
+### Examples
+
+Provide examples under the `Examples` section. This helps users understand how to use the function or class. Use triple backticks to enclose the example code.
+
+Example:
+
+```python
+def example_function(param1, param2):
+    """This function performs an example operation.
+
+    Examples:
```

Comment:
This lacks a bit of clarity. Should I use triple backticks in each line of code? Only at the start?

Reply:
see comment above